yaptide/simulation_runner/shieldhit_runner.py

```diff
@@ -10,108 +10,91 @@
     with tempfile.TemporaryDirectory() as tmp_output_path:
 
-        for config_filename in input_dict:
-            abs_input_path = os.path.join(tmp_output_path, config_filename)
-            with open(abs_input_path, 'w') as temp_input_file:
-                temp_input_file.write(input_dict[config_filename])
+        # digest dictionary with project data (extracted from JSON file)
+        # and generate SHIELD-HIT12A input files
+        conv_runner = ConvRunner(parser=DummmyParser(),
+                                 input_data=raw_input_dict,
+                                 output_dir=tmp_output_path)
+
+        conv_runner.run_parser()
 
         settings = SimulationSettings(input_path=tmp_output_path,
                                       simulator_exec_path=None,
                                       cmdline_opts='')
 
-        runner_obj = Runner(jobs=param_dict['jobs'],
-                            keep_workspace_after_run=False,
-                            output_directory=tmp_output_path)
+        runner_obj = SHRunner(jobs=param_dict['jobs'],
+                              keep_workspace_after_run=False,
+                              output_directory=tmp_output_path)
 
         start_time = timeit.default_timer()
+
         isRunOk = runner_obj.run(settings=settings)
         if not isRunOk:
             return None
 
         elapsed = timeit.default_timer() - start_time
         print("MC simulation took {:.3f} seconds".format(elapsed))
 
-        estimator = runner_obj.get_data()
-
-        return estimator
-
-
-if __name__ == '__main__':
-    sys.exit(run_shieldhit(sys.argv[1:]))
+        estimators_dict = runner_obj.get_data()
+
+        return dummy_convert_output(estimators_dict)
+
+
+def dummy_convert_output(estimators_dict):
+    """Dummy function for converting simulation output to dictionary"""
+    if not estimators_dict:
+        return {"message": "No estimators"}
+
+    # result_dict is the dictionary object, which is later converted to json
+    # to provide readable api response for fronted
+
+    # result_dict contains the list of estimators
+    result_dict = {"estimators": []}
+    for est_name in estimators_dict:
+
+        # est_dict contains list of pages
+        est_dict = {
+            "name" : est_name,
+            "pages": []}
+        for page in estimators_dict[est_name].pages:
+
+            page_dim = page.dimension
+
+            # handling 1 dimension page
+            if page_dim == 1:
+                axis = page.plot_axis(0)
+                x_values = axis.data
+                y_values = page.data_raw.flatten()
+
+                # for 1 dimension page, dict contains:
+                # "dimensions" indicating it is 1 dim page
+                # "unit"
+                # "x_values" which is list of x values
+                # "y_values" which is list of y values
+                page_dict = {
+                    "dimensions" : page_dim,
+                    "unit": str(axis.unit),
```

Comment:
What about axis name ?
How frontend should know whether we want to have X,Y or Z on horizontal axis ?

Reply:
The problem I found was that, the axis didn't actually have any name but we will discuss it on the meeting

Reply:
Even if the name is empty, use it. Fix in pymchelper is incoming https://github.com/DataMedSci/pymchelper/pull/477

Reply:
Fixed in 1.9.1 https://github.com/DataMedSci/pymchelper/releases

Reply:
Added `name`